clang/test/Interpreter/lambda.cpp

```diff
@@ -14,4 +15,13 @@ auto r2 = l2();
 auto r3 = l2();
 // CHECK: TWO
 
-%quit
+// Verify non-local lambda capture error is correctly reported
+int x = 42;
+
+// expected-error {{non-local lambda expression cannot have a capture-default}}
+auto capture = [&]() { return x * 2; };
+
```

Comment:
Don’t we want to check the value of X here to be 84?

Reply:
Sorry, I think I might have missed this. 

If I remember correctly, the above is an expected error 
```
clang-repl> int x = 42; auto capture = [&]() { return x * 2; };
In file included from <<< inputs >>>:1:
input_line_1:1:28: error: non-local lambda expression cannot have a capture-default
    1 | int x = 42; auto capture = [&]() { return x * 2; };
      |                            ^
error: Parsing failed.
```
So we wouldn't have to check for the value to be 84 here !

Reply:
I think this is all that we need here. Should be ready. let me try updating !